### PR TITLE
ProposedTransaction.__init__ now convert tag value to a Tag object

### LINKED ISSUE #100: ProposedTransaction.__init__ should convert `tag` value to a `Tag` object
TryteString.from_string has bad padding. If you run:

```
tag = TryteString.from_string("tagname")
```
It won't return a string that is 27 characters long, which creates issues with other functions. Need to make ugly hacks, that add 9's up to the 27 lenght requirement.

Example structure it was generating issues with:

```
transfers=[

                ProposedTransaction(

                    address=Address("JEMEHHS9EEASH9UFOLYAKBCTNKGBADCLTKCEOEKPZVILIGTZOGYMFMWKQABVVNKHGHBHTJNRVDDMMUQOX"),

                    value=0

                ),

                ProposedTransaction(

                    address=Address("LORKNNVSBTILWAUZBK9RTMVFPJOWAU9KOIINNGYISJQ9P9AFBFH9OJNXBILBFRPVHGVNIBB9LYGUHCESB"),

                    value=0,

                    message=TryteString.from_string("First transaction"),

                    tag=TryteString.from_string("xCarx")

                )

    ],
```

### PR DESCRIPTION
**Pull request by rpitonak**
- Fixes #100 
- iota/transaction/creation.py now conforms to PEP8

I will be glad for the comments and suggestions.

### CODE REVIEW

File: iota/transaction/creation.py [hunk 1]
```diff
@@ -104,365 +105,367 @@ def increment_legacy_tag(self):
...
+                ),
+            )
+
+        # Generate bundle hash.
+        while True:
+            sponge = Kerl()
+            last_index = len(self) - 1
+
+            # type: Tuple[int, ProposedTransaction]
+            for (i, txn) in enumerate(self):
```
Comment (todofixthis) [inline iota/transaction/creation.py]: ❔ Does the type hint work when it's on the line preceding the `for` loop?  I thought it had to go on the same line.

Comment (todofixthis) [reply iota/transaction/creation.py]: Example:
Type hint placed on the previous line — note that IDE autocomplete doesn't work:
type hint on previous line - no autocomplete

Type hint placed at the end of the line — now IDE autocomplete works as expected.
type hint at end of line - autocomplete works

**Comment by rpitonak** [comment]
Hey @todofixthis thank you for the review. The type hint is already fixed.

I will send another commit with Unit test.

**Comment by todofixthis** [comment]
Hey Radoslav.  Because this PR also includes PEP-8 formatting changes, could you also follow-up with the developers working on #147 to make sure that any conflicts can be resolved?

**Pull request merged**